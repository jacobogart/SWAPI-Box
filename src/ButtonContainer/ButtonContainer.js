import React from 'react';

export default function CardContainer(props) {
  return (
    <section 
      className="ButtonContainer"
      onClick={event => props.selectCategory(event)}
    >
      <button className="category-button">people</button>
      <button className="category-button">planets</button>
      <button className="category-button">vehicles</button>
    </section>
  );
}