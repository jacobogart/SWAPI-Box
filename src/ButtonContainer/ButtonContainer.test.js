import React from "react";
import { shallow } from "enzyme";
import ButtonContainer from './ButtonContainer';


describe('ButtonContainer', () => {
  let mockSelectCategory = jest.fn();
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(
    <ButtonContainer
      selectCategory={mockSelectCategory}
    />)
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should invoke selectCategory on click', () => {
    wrapper.find(".ButtonContainer").simulate('click');
    expect(mockSelectCategory).toHaveBeenCalled();
  })
})