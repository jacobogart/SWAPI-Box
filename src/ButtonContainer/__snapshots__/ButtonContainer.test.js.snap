// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonContainer should match snapshot 1`] = `ShallowWrapper {}`;
